overlord/devicestate/devicemgr.go

```diff
@@ -849,7 +849,21 @@ func (m *DeviceManager) RequestSystemAction(systemLabel string, action SystemAct
 		return ErrUnsupportedAction
 	}
 
-	// TODO:UC20 update boot environment and schedule a reboot
+	m.state.Lock()
+	defer m.state.Unlock()
+
+	deviceCtx, err := DeviceCtx(m.state, nil, nil)
+	if err != nil && err != state.ErrNoState {
```

Comment:
it doesn't look correct to ignore ErrNoState here, we'll crash below no?

Reply:
Yeah. Before, the bits in boot would complain about device being nil, but those were dropped subsequent patches.